pyomo/contrib/edi/formulation.py

```diff
@@ -0,0 +1,455 @@
+                        )
+                    if size[i] == 1:
+                        raise ValueError(
+                            'A value of 1 is not valid for defining size.  Use fewer dimensions.'
+                        )
+                    if i == 0:
+                        st = Set(initialize=list(range(0, size[i])))
+                    else:
+                        st *= Set(initialize=list(range(0, size[i])))
+                st.construct()
+                self.add_component(
+                    name,
+                    Param(
+                        st,
+                        name=name,
+                        initialize=value,
+                        within=within,
+                        doc=description,
+                        units=decodeUnits(units),
+                        mutable=True,
+                    ),
+                )
+            else:
+                if isinstance(size, int):
+                    if size == 1 or size == 0:
+                        self.add_component(
+                            name,
+                            Param(
+                                name=name,
+                                initialize=value,
+                                within=within,
+                                doc=description,
+                                units=decodeUnits(units),
+                                mutable=True,
+                            ),
+                        )
+                    else:
+                        st = Set(initialize=list(range(0, size)))
+                        st.construct()
+                        self.add_component(
+                            name,
+                            Param(
+                                st,
+                                name=name,
+                                initialize=value,
+                                within=within,
+                                doc=description,
+                                units=decodeUnits(units),
+                                mutable=True,
+                            ),
+                        )
+                else:
+                    raise ValueError(
+                        'Invalid size.  Must be an integer or list/tuple of integers'
+                    )
+        else:
+            self.add_component(
+                name,
+                Param(
+                    name=name,
+                    initialize=value,
+                    within=within,
+                    doc=description,
+                    units=decodeUnits(units),
+                    mutable=True,
+                ),
+            )
+
+        self.__dict__[name].construct()
+        self._constant_keys.append(name)
+        return self.__dict__[name]
+
+    def Objective(self, expr, sense=minimize):
+        self._objective_counter += 1
+        self.add_component(
+            'objective_' + str(self._objective_counter),
+            Objective(expr=expr, sense=sense),
```

Comment:
Scoping issue with multiple definitions of `Objective`?

Reply:
Recommended fix: use `pyo.Objective` in line 296

Reply:
Corrected in new commit